cdap-security/src/main/java/co/cask/cdap/security/store/FileSecureStore.java

```diff
@@ -102,39 +99,26 @@
    * @param name Name of the element to store.
    * @param data The data that needs to be securely stored.
    * @param description User provided description of the entry.
-   * @param properties Metadata associated with the data  @throws IOException
+   * @param properties Metadata associated with the data
+   * @throws IOException If it failed to store the key in the store.
    */
   @Override
   public void put(String namespace, String name, byte[] data, String description, Map<String, String> properties)
     throws IOException {
     String keyName = getKeyName(namespace, name);
     SecureStoreMetadata meta = SecureStoreMetadata.of(name, description, properties);
     SecureStoreData secureStoreData = new SecureStoreData(meta, data);
-    Key key = null;
     writeLock.lock();
     try {
       if (keyStore.containsAlias(keyName)) {
-        key = keyStore.getKey(keyName, password);
+        throw new IOException("The key " + name + " under namespace " + namespace + "already exists.");
```

Comment:
I think I will have to do this as part of the authorization task any way, so I will make the change in that PR.


> On Jul 18, 2016, at 3:50 PM, Bhooshan Mogal <notifications@github.com> wrote:
> 
> In cdap-security/src/main/java/co/cask/cdap/security/store/FileSecureStore.java <https://github.com/caskdata/cdap/pull/6066#discussion_r71245515>:
> 
> >      writeLock.lock();
> >      try {
> >        if (keyStore.containsAlias(keyName)) {
> > -        key = keyStore.getKey(keyName, password);
> > +        throw new IOException("The key " + name + " under namespace " + namespace + "already exists.");
> should create a class called SecureKey that extends EntityId and implements NamespacedId in cdap-proto, and throw a NotFoundException with an object of that class here. The REST APIs will then respond with the right code automatically
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/caskdata/cdap/pull/6066/files/96308b540a6cda9ab5ee33d275c4f304922e19e0#r71245515>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AAFGk7bK-LZP1_Kv4arCGbFvv1KrDyHCks5qXAM2gaJpZM4JOEs0>.
>